From 933ba1cb66d124454889c8c2b844fee2b0e0960b Mon Sep 17 00:00:00 2001
From: Yangyu Chen <cyy@cyyself.name>
Date: Tue, 26 Oct 2021 15:40:38 +0800
Subject: [PATCH 068/124] fixed UB which lead to cache coherence bug

---
 arch/riscv/kernel/sbi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/riscv/kernel/sbi.c b/arch/riscv/kernel/sbi.c
index 4daea631c078..77132e824954 100644
--- a/arch/riscv/kernel/sbi.c
+++ b/arch/riscv/kernel/sbi.c
@@ -560,7 +560,7 @@ void sbi_dma_sync(unsigned long start,
 	register unsigned long i asm("a0") = start & ~(L1_CACHE_BYTES - 1);
 
 	for (; i < ALIGN(start + size, L1_CACHE_BYTES); i += L1_CACHE_BYTES)
-		__asm__ __volatile__(".long 0x02b5000b");
+		__asm__ __volatile__(".long 0x02b5000b" : : "r"(i));
 
 	__asm__ __volatile__(".long 0x01b0000b");
 #endif
-- 
2.20.1

