From 8255fd9e0ea6e7c0d621f401fc6fbc7fdfc73ab2 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Mon, 15 Nov 2021 23:38:00 -0600
Subject: [PATCH 070/124] riscv: Allow suspend

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 arch/riscv/Kconfig | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/riscv/Kconfig b/arch/riscv/Kconfig
index fb462acf769b..2d996d6f4265 100644
--- a/arch/riscv/Kconfig
+++ b/arch/riscv/Kconfig
@@ -194,6 +194,9 @@ config ARCH_WANT_GENERAL_HUGETLB
 config ARCH_SUPPORTS_UPROBES
 	def_bool y
 
+config ARCH_SUSPEND_POSSIBLE
+	def_bool y
+
 config STACKTRACE_SUPPORT
 	def_bool y
 
-- 
2.20.1

