--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -761,6 +761,11 @@ my %targets = (
         perlasm_scheme   => "linux64",
     },
 
+    "linux-riscv64-openwrt" => {
+        inherit_from     => [ "linux-generic64"],
+        perlasm_scheme   => "linux64",
+    },
+
     #### IA-32 targets...
     #### These two targets are a bit aged and are to be used on older Linux
     #### machines where gcc doesn't understand -m32 and -m64
